src/OrbitQt/CallTreeWidget.cpp

```diff
@@ -170,28 +170,87 @@ void CallTreeWidget::ResizeColumnsIfNecessary() {
   column_resizing_state_ = ColumnResizingState::kDone;
 }
 
-static std::string BuildStringFromIndices(const QModelIndexList& indices) {
-  std::string buffer;
+static std::string BuildStringFromIndices(QTreeView* tree_view, const QModelIndexList& indices) {
+  // The order of rows in the selection model is not the order in which the rows are displayed.
+  // We can recover that order by sorting using as comparison function for two rows the
+  // lexicographic order of the vector of the row indices from the root to those rows.
+  struct ItemWithAncestorRows {
+    enum ExpandedState { kLeaf = 0, kExpanded, kCollapsed };
+
+    std::string data;
+    bool is_first_in_row = false;
+    ExpandedState expanded_state = kLeaf;
+    std::list<int> ancestor_rows;
```

Comment:
As far as I can see, a forward_list should be enough. But that's kind of nitty...

Reply:
You had convinced me, but I need `size` below, for the indentation.

Reply:
Ahhh. I see. I wasn't aware that forward_list doesn't have size.